From 91f138e1389ed891a343532d269a7cb55a983c5d Mon Sep 17 00:00:00 2001
From: Dan Dennedy <dan@dennedy.org>
Date: Sat, 8 May 2021 18:09:35 -0700
Subject: [PATCH] fix #1089 JACK audio broken with MLT7

---
 src/mltcontroller.cpp | 12 ++++++------
 src/mltcontroller.h   |  4 ++--
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/mltcontroller.cpp b/src/mltcontroller.cpp
index c0d854f2b..c4fdc9609 100644
--- a/src/mltcontroller.cpp
+++ b/src/mltcontroller.cpp
@@ -272,10 +272,10 @@ void Controller::stop()
     stopJack();
 }
 
-void Controller::on_jack_started(mlt_properties, void* object, const mlt_position *position)
+void Controller::on_jack_started(mlt_properties, void* object, mlt_event_data data)
 {
-    if (object && position)
-        (static_cast<Controller*>(object))->onJackStarted(*position);
+    if (object)
+        (static_cast<Controller*>(object))->onJackStarted(Mlt::EventData(data).to_int());
 }
 
 void Controller::onJackStarted(int position)
@@ -287,10 +287,10 @@ void Controller::onJackStarted(int position)
     }
 }
 
-void Controller::on_jack_stopped(mlt_properties, void* object, const mlt_position *position)
+void Controller::on_jack_stopped(mlt_properties, void* object, mlt_event_data data)
 {
-    if (object && position)
-        (static_cast<Controller*>(object))->onJackStopped(*position);
+    if (object)
+        (static_cast<Controller*>(object))->onJackStopped(EventData(data).to_int());
 }
 
 void Controller::onJackStopped(int position)
diff --git a/src/mltcontroller.h b/src/mltcontroller.h
index 436b2a0b3..574ef0175 100644
--- a/src/mltcontroller.h
+++ b/src/mltcontroller.h
@@ -190,9 +190,9 @@ class Controller
     QString m_projectFolder;
     QMutex m_saveXmlMutex;
 
-    static void on_jack_started(mlt_properties owner, void* object, const mlt_position *position);
+    static void on_jack_started(mlt_properties owner, void* object, mlt_event_data data);
     void onJackStarted(int position);
-    static void on_jack_stopped(mlt_properties owner, void* object, const mlt_position *position);
+    static void on_jack_stopped(mlt_properties owner, void* object, mlt_event_data data);
     void onJackStopped(int position);
     void stopJack();
 };
